src/nimblepkg/reversedeps.nim

```diff
@@ -76,6 +76,24 @@ proc getRevDeps*(options: Options, pkg: PackageInfo): seq[PkgTuple] =
 
       result.add(pkgTup)
 
+proc contains(pkgs: seq[PackageInfo], pkg: PackageInfo): bool =
+  for lpkg in pkgs:
+    if lpkg.name == pkg.name and lpkg.myPath == pkg.myPath:
+      return true
+
+proc getAllRevDeps*(options: Options, pkg: PackageInfo, result: var seq[PackageInfo]) =
+  if pkg in result:
```

Comment:
For very large lists of packages this could get pretty expensive. It might be worth using a ``HashSet[PackageInfo]``.

Reply:
Not done yet, will ping when done.

Reply:
I personally don't see the value of this effort - even uninstalling 25 packages runs almost instantly.

```
> timeit nimble uninstall c2nim -i -n
    Looking for c2nim (any version)
   Checking reverse dependencies
    Prompt: The following packages will be removed:
        ...   treesitter_c (#head), treesitter_bash (#head), treesitter_cpp (#head), treesitter_css (#head), treesitter_go (#head), treesitter_html (#head), treesitter_javascript (#head), treesitter_ocaml (#head), treesitter_php (#head), treesitter_python (#head), treesitter_ruby (#head), treesitter_rust (#head), treesitter_typescript (#head), treesitter_agda (#head), treesitter_c_sharp (#head), treesitter_haskell (#head), treesitter_java (#head), treesitter_scala (#head), treesitter (#head), nimarchive (0.2.0), libsvm (0.1.3), nimrax (0.1.1), nimgen (0.5.0), c2nim (#3ec45c24585ebaed)
        ... Do you wish to continue? -> [forced no]

Version Number:   Windows NT 6.2 (Build 9200)
Exit Time:        6:03 pm, Monday, January 7 2019
Elapsed Time:     0:00:00.499
Process Time:     0:00:00.015
System Calls:     58650
Context Switches: 4006
Page Faults:      9529
Bytes Read:       3087857
Bytes Written:    29437
Bytes Other:      1014221
```

Reply:
Okay, we can always optimise this later.